config/configgrpc/configgrpc.go

```diff
@@ -236,7 +237,7 @@ func (gss *GRPCServerSettings) ToListener() (net.Listener, error) {
 }
 
 // ToServerOption maps configgrpc.GRPCServerSettings to a slice of server options for gRPC
-func (gss *GRPCServerSettings) ToServerOption() ([]grpc.ServerOption, error) {
+func (gss *GRPCServerSettings) ToServerOption(extensions map[string]component.ServiceExtension) ([]grpc.ServerOption, error) {
```

Comment:
Can you try something different?

* Create an interface somewhere named, say, "WantsServiceExtensions", defining a `SetServiceExtensions(map[string]component.ServiceExtension` function
* In the bootstrap code for the `application`, where it builds the `config` objects, check whether the config that was just built is a `WantsServiceExtensions`, and if so, call `SetServiceExtensions` with the map of extensions.

This way, we have a dependency injection-like mechanism, abstracting this requirement away from the callers, making it easy to use this config with all receivers without them having to manually wire up the map.

Reply:
Oh interesting. I might be looking at the wrong place in code, but won't that be earlier than the extensions are built?

Reply:
I don't remember how the code looks like, but if they are part of the same func/chain, you can iterate over the config objects after you build the extensions.

Reply:
Here https://github.com/open-telemetry/opentelemetry-collector/blob/7eb262845a4ce73782fa967138e7859af4217d9f/service/service.go#L361 extension and receivers are already built, you can call SetServiceExtensions. I would do it uniformly for all components, not just receivers.

Reply:
Pushed a new commit using this approach, let me know what you think.